### PR TITLE
Allow use of headless hostname

### PR DESCRIPTION
**Pull request by angelini**
One issue with using `dateilager.gadget-services.net` to resolve a DL instance is that the K8S service will resolve this to a single instance IP.

If for wtv reason you fail to connect to that IP, for example during a DL deployment, it cannot recover by trying another instance because it has already been resolved to that fixed IP.

If instead we use a headless service, we can learn about all instances and then retry among different ones.

We already make use of this for our GRPC JS client, this just adds it to the binary client.

### CODE REVIEW

**Comment by angelini** [comment]
I have 🎩 this from evil-ssh pod and I was able to connect to DL using the headless hostname and omitting it.

**Comment by angelini** [comment]
Just fixing up the makefile, missed a few changes there

File: pkg/client/client.go [hunk 1]
```diff
@@ -55,7 +55,8 @@ func NewClientConn(conn *grpc.ClientConn) *Client {
 }
 
 type options struct {
-	token string
+	headlessAdress string
```
Comment (thegedge) [inline pkg/client/client.go]: nit: typo on address

**Review by thegedge — APPROVED** [review]
🚀

**Review by MikePresman — APPROVED** [review]
LGTM

**Pull request merged**